File: src/ClaudeToolbar.Core/Formatting/BarLevel.cs
namespace ClaudeToolbar.Core.Formatting;

public enum BarLevel
{
    Ok,
    Warn,
    Crit,
}

public static class BarLevelResolver
{
    public static BarLevel Resolve(double utilization, int warnThreshold, int critThreshold)
    {
        if (utilization < warnThreshold) return BarLevel.Ok;
        if (utilization < critThreshold) return BarLevel.Warn;
        return BarLevel.Crit;
    }
}

File: mac/Sources/ClaudeToolbarCore/Formatting/BarLevel.swift
public enum BarLevel: Sendable, Equatable {
    case ok
    case warn
    case crit
}

public enum BarLevelResolver {
    public static func resolve(utilization: Double, warnThreshold: Int, critThreshold: Int) -> BarLevel {
        if utilization < Double(warnThreshold) { return .ok }
        if utilization < Double(critThreshold) { return .warn }
        return .crit
    }
}
